tests/gdx-tests/src/com/badlogic/gdx/tests/SpriteBatchPerformanceTest.java

```diff
@@ -0,0 +1,88 @@
+/*******************************************************************************
+ * Copyright 2011 See AUTHORS file.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package com.badlogic.gdx.tests;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.graphics.GL20;
+import com.badlogic.gdx.graphics.Texture;
+import com.badlogic.gdx.graphics.g2d.BitmapFont;
+import com.badlogic.gdx.graphics.g2d.SpriteBatch;
+import com.badlogic.gdx.math.WindowedMean;
+import com.badlogic.gdx.tests.utils.GdxTest;
+import com.badlogic.gdx.tests.utils.GdxTestConfig;
+import com.badlogic.gdx.utils.StringBuilder;
+
+@GdxTestConfig
+public class SpriteBatchPerformanceTest extends GdxTest {
+
+	private Texture texture;
+	private SpriteBatch spriteBatch;
+	private WindowedMean counter = new WindowedMean(10000);
+	private StringBuilder stringBuilder = new StringBuilder();
+
+	private BitmapFont bitmapFont;
+
+	@Override
+	public void create () {
+		texture = new Texture(Gdx.files.internal("data/badlogic.jpg"));
+		spriteBatch = new SpriteBatch(8191);
+		bitmapFont = new BitmapFont();
+	}
+
+	@Override
+	public void render () {
+		Gdx.gl20.glViewport(0, 0, Gdx.graphics.getBackBufferWidth(), Gdx.graphics.getBackBufferHeight());
+		Gdx.gl20.glClearColor(0.2f, 0.2f, 0.2f, 1);
+		Gdx.gl20.glClear(GL20.GL_COLOR_BUFFER_BIT);
+
+		spriteBatch.begin();
+
+		// Accelerate the draws
+		for (int j = 0; j < 100; j++) {
+
+			// fill the batch
+			for (int i = 0; i < 8190; i++) {
+				spriteBatch.draw(texture, 0, 0, 1, 1);
```

Comment:
Minor thing but maybe instead of drawing a single point we can change it to something like 20px * 20px just for the user to see something is happening.

Reply:
@obigu  i think it would bias the test since it will overload fragment shader stage, considering there are about 819.000 drawings, it would have to render 20x20 pixels each, so ~327 Mega pixels which could be a lot compared to the actual amount (less than 1 Mega pixels). In this end it would defeat isolation of this test.

Reply:
if we want to isolate a bit more, we could disable blending and use depth test (never pass) in order to fully bypass fragment stage.